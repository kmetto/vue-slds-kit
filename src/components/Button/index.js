import install from '../install';
import component from './Button.vue';

export default install(component);
