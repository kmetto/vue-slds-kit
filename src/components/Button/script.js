export default {
  name: 'slds-button',
  props: {
    disabled: {
      type: Boolean,
      default: false,
    },
    neutral: {
      type: Boolean,
      default: false,
    },
    brand: {
      type: Boolean,
      default: false,
    },
    inverse: {
      type: Boolean,
      default: false,
    },
    destructive: {
      type: Boolean,
      default: false,
    },
    success: {
      type: Boolean,
      default: false,
    },
  },
  data() {
    return {
      classes: {
        'slds-button_neutral': this.neutral,
        'slds-button_brand': this.brand,
        'slds-button_inverse': this.inverse,
        'slds-button_destructive': this.destructive,
        'slds-button_success': this.success,
      },
    };
  },
};
