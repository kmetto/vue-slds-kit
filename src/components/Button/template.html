<button class="slds-button" :class="classes" :disabled="disabled">
  <slot name="leftIcon"></slot>
  <slot></slot>
  <slot name="rightIcon"></slot>
</button>