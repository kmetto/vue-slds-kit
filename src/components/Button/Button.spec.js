import { mount, shallow } from '@vue/test-utils';
import { createRenderer } from 'vue-server-renderer';
import Button from './Button.vue';

describe('Button', () => {
  test('must be an instance of Vue', () => {
    const wrapper = mount(Button);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('matches snapshot', () => {
    const renderer = createRenderer();
    const wrapper = shallow(Button);
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw err;
      }
      expect(str).toMatchSnapshot();
    });
  });

  test('matches snapshot with slot', () => {
    const renderer = createRenderer();
    const wrapper = shallow(Button, {
      slots: {
        default: 'Button',
      },
    });
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw err;
      }
      expect(str).toMatchSnapshot();
    });
  });

  test('displays slds-button_neutral class', () => {
    const wrapper = mount(Button, {
      propsData: {
        neutral: true,
      },
    });
    expect(wrapper.contains('button.slds-button.slds-button_neutral')).toBeTruthy();
  });

  test('displays slds-button_brand class', () => {
    const wrapper = shallow(Button, {
      propsData: {
        brand: true,
      },
    });
    expect(wrapper.contains('button.slds-button.slds-button_brand')).toBeTruthy();
  });

  test('displays slds-button_inverse class', () => {
    const wrapper = shallow(Button, {
      propsData: {
        inverse: true,
      },
    });

    expect(wrapper.contains('button.slds-button.slds-button_inverse')).toBeTruthy();
  });

  test('displays slds-button_destructive class', () => {
    const wrapper = shallow(Button, {
      propsData: {
        destructive: true,
      },
    });

    expect(wrapper.contains('button.slds-button.slds-button_destructive')).toBeTruthy();
  });

  test('displays slds-button_success class', () => {
    const wrapper = shallow(Button, {
      propsData: {
        success: true,
      },
    });

    expect(wrapper.contains('button.slds-button.slds-button_success')).toBeTruthy();
  });

  test('displays disabled button', () => {
    const wrapper = shallow(Button, {
      propsData: {
        disabled: true,
      },
    });

    expect(wrapper.attributes().disabled).toBeTruthy();
  });
});
