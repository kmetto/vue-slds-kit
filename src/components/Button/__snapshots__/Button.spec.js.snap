// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `"<button data-server-rendered=\\"true\\" class=\\"slds-button\\">  </button>"`;

exports[`Button matches snapshot with slot 1`] = `"<button data-server-rendered=\\"true\\" class=\\"slds-button\\"> Button </button>"`;
